src/main/java/frc/trigon/robot/subsystems/swerve/SwerveModule.java

```diff
@@ -0,0 +1,128 @@
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import frc.trigon.robot.utilities.Conversions;
+
+public class SwerveModule {
+    private final WPI_TalonSRX angleEncoder;
+    private final WPI_TalonFX angleMotor;
+    private final WPI_TalonFX driveMotor;
+
+    private final double encoderOffset;
+    private SwerveModuleState targetState;
+
+    public SwerveModule(SwerveModuleConstants moduleConstants) {
+        this.angleEncoder = moduleConstants.angleEncoder;
+        this.angleMotor = moduleConstants.angleMotor;
+        this.driveMotor = moduleConstants.driveMotor;
+        this.encoderOffset = moduleConstants.encoderOffset;
+
+        angleMotor.configSelectedFeedbackSensor(FeedbackDevice.RemoteSensor0);
+        configRemoteSensor();
+    }
+
+    public void setTargetState(SwerveModuleState targetState,boolean isOpenLoop) {
+        this.targetState = targetState;
+        optimizeTargetState();
+        setMotorsToStates(
+                isOpenLoop,
+                targetState.angle.getDegrees(),
+                targetState.speedMetersPerSecond
+        );
+    }
+
+    public SwerveModuleState getCurrentState() {
+        return new SwerveModuleState(getVelocity(), getModuleAngle());
+    }
+
+    private void optimizeTargetState() {
+        double scoped = scope(targetState.angle.getDegrees());
+        double flipped = scope(180 + targetState.angle.getDegrees());
+        double scopeDiff = Math.abs(scoped - getDegrees());
+        double flippedDiff = Math.abs(flipped - getDegrees());
+        if(scopeDiff < flippedDiff)
+            targetState.angle = Rotation2d.fromDegrees(scoped);
+        else {
+            targetState.angle = Rotation2d.fromDegrees(flipped);
+            targetState.speedMetersPerSecond *= -1;
+        }
+    }
+
+    public double scope(double targetAngle) {
+        double rawCurrentAngle = getDegrees() % 360;
+        double rawTargetAngle = targetAngle % 360;
+        double difference = rawTargetAngle - rawCurrentAngle;
+
+        if(difference < -180)
+            difference += 360;
+
+        else if(difference > 180)
+            difference -= 360;
+
+        return difference + getDegrees();
+    }
+
+    public void setMotorsToStates(boolean isOpenLoop ,double targetAngle ,double targetVelocity) {
+        setTargetAngle(targetAngle);
+        setTargetDriveVelocity(isOpenLoop ,targetVelocity);
+    }
+
+    public void setTargetAngle(double targetAngle) {
+        double targetAnglePosition = Conversions.degreesToMagTicks(targetAngle);
+        angleMotor.set(ControlMode.Position, targetAnglePosition);
+    }
+
+    public void setTargetDriveVelocity(boolean isOpenLoop , double targetVelocity){
+        if(isOpenLoop)
+            setTargetDriveMotorPower(targetVelocity);
+        else setTargetDriveMotorVelocity(targetVelocity);
+
+    }
+    public void setTargetDriveMotorPower(double targetPower){
```

Comment:
it's not target, you're actually setting the power
and the word "motor" is redundant